Source/JavaScriptCore/API/JSContext.h

```diff
@@ -57,6 +57,15 @@ JSC_CLASS_AVAILABLE(macos(10.9), ios(7.0))
 */
 - (instancetype)initWithVirtualMachine:(JSVirtualMachine *)virtualMachine;
 
+/*!
+@method
+@abstract Create a JSContext in the specified virtual machine and context.
+@param virtualMachine The JSVirtualMachine in which the context will be created.
+@param context JSGlobalContextRef the context will hold.
+@result The new context.
+*/
+- (instancetype)initWithVirtualMachine:(JSVirtualMachine *)virtualMachine context:(JSGlobalContextRef)context NS_DESIGNATED_INITIALIZER NS_UNAVAILABLE;
```

Comment:
This adds a public API, which is a very significant change. It would block the PR on going through certain Apple internal review first.

Reply:
Xcode refuses to compile this any other way but I will see if I could find another solution

Reply:
I think this method should be declared in JSContextInternal.h. That header is not exposed to clients of JSC.

Reply:
Tried that. Xcode will not build.

Reply:
Yeah, it needs to be inside the `@implementation JSContext (Internal)` block.

Reply:
definition or just the declaration or both?

Reply:
If the declaration is inside the `@interface JSContext(Internal)` then the implantation needs to be inside the `@implementation JSContext (Internal)`. At least if I remember correctly.

Reply:
Not working sadly. Only the way I had it before works.

Reply:
@kmiller68 So we are at an impasse. Making it public is the only way this even builds

Reply:
You're trying to add new public API to fix... a build failure? That did not exist before your pull request?

🤔️

Reply:
Yeah. I know...

Reply:
> You're trying to add new public API to fix... a build failure? That did not exist before your pull request?
> 
> 🤔️

Yeah just read the build-logs.

Reply:
I fixed it. @mcatanzaro